### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by florianl**
Signed-off-by: Florian Lehner <dev@der-flo.net>

This PR adjusted error handling and makes sure errors are not ignored.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -41,11 +41,9 @@ func init() {
 	rootCmd.PersistentFlags().BoolVar(&noExpand, "no-expand", false, "expand files indices")
 	rootCmd.MarkFlagRequired("record")
 	rootCmd.PersistentFlags().StringVarP(&server, "server", "s", "http://opendata.cern.ch", "CERN Open Data server to query")
-	err := doc.GenMarkdownTree(rootCmd, "docs")
-	if err != nil {
-		fmt.Errorf("error generating docs", err)
```
Comment (florianl) [inline cmd/root.go:46]: `fmt.Errorf()` is a pure function but its return value is ignored. So I have replaced it with `panic()`. An alternative would be `fmt.Fprintf(os.Stderr, ...)` & `os.Exit(1)` to let the user know about the issue. Let me know, if you prefer the alternative.

**Comment by florianl** [comment]
No sure if the github action failed because of the github incident at the time [0]. The error message of `goreleaser` mentiones `Error: Username and password required`.

[0] https://www.githubstatus.com/incidents/bdbzpz7qxmbx

**Comment by clelange** [comment]
Hi Flo,

I think the failure is my fault, since I never set up goreleaser to work with pull requests. Sorry about taking longer to get back to you. I'm wondering whether you are actively using this tool or are planning to do so, since I've only created it to get something to work for a CMS Open Data workshop last year (and I really like small binaries). The official tool, which also has more features but is written in Python, can be found at https://github.com/cernopendata/cernopendata-client.
In any case, I'll have a closer look at your PRs and will probably merge them.

**Comment by florianl** [comment]
Hi Clemens,

I started to (and still to) explore data provided at http://opendata.cern.ch/. While https://github.com/cernopendata/cernopendata-client provides more features, for my use case this project is perfect. And it is easier (at least for me) to integrate it into a data workflow pipeline.

File: cmd/root.go (hunk 1 above)
Comment (clelange) [reply cmd/root.go:46]: I think that's fine, it will still print the error as far as I understand.

**Review by clelange — APPROVED** [review]
Really happy to hear this is useful for you!

**Pull request merged**